src/st/no0/4C750.c

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+#include "common.h"
+#include "no0.h"
+
+// INCLUDE_ASM("st/no0/nonmatchings/4C750", func_us_801CC750);
+
+extern void func_us_801CC8F8(Entity*);
+extern u16 D_us_80180A88;
+
+void func_us_801CC750(Entity* self) {
+    Entity* entityPtr;
+    s16 i;
+    Primitive* prim;
+    s32 primIndex;
+
+    if (self->step != 0) {
+        return;
+    }
+
+    InitializeEntity(&D_us_80180A88);
+    primIndex = g_api.AllocPrimitives(PRIM_GT4, 9);
+    if (primIndex != -1) {
+        prim = &g_PrimBuf[primIndex];
+        i = -0x10;
+        self->primIndex = primIndex;
+        self->flags |= FLAG_HAS_PRIMS;
+        while (prim != NULL) {
+            prim->tpage = 0xF;
+            prim->clut = 0x2A;
+            prim->u0 = prim->u2 = 0xC0;
+            prim->u1 = prim->u3 = 0xE0;
+            prim->v0 = prim->v1 = 0x80;
+            prim->v2 = prim->v3 = 0xB0;
+            prim->x0 = prim->x2 = i;
+            prim->x1 = prim->x3 = i + 0x20;
+            prim->y0 = prim->y1 = 0x91;
+            prim->y2 = prim->y3 = 0xC1;
+            prim->priority = 0;
+            prim->drawMode = DRAW_DEFAULT;
+            prim = prim->next;
+            i += 0x1E;
+        }
+    }
+
+    entityPtr = (Entity*)((char*)self + 0xBC);
+
+    for (i = -0x10; i < 0x130; i += 0x60) {
+        DestroyEntity(entityPtr);
+        entityPtr->entityId = 0x16;
+        entityPtr->pfnUpdate = func_us_801CC8F8;
+        entityPtr->posY.i.hi = 0x8E;
+        entityPtr->posX.i.hi = i;
+
+        entityPtr = (Entity*)((char*)entityPtr + 0xBC);
```

Comment:
Looks like entityPtr++?

Reply:
I'm stuck with this a bit. Appreciate the help

Reply:
I don't think it's enityPtr++, it looks like we're jumping it forward 0xBC.

perhaps
`entityPtr += 0xBC;`

However, the weird cast might make that not correct.  But give that a try.

Reply:
sizeof(Entity) is 0xBC https://github.com/Xeeynamo/sotn-decomp/blob/c1b36a102974e1264733e54afec4019b679cdc04/include/game.h#L858

I'm pretty sure it's something like this https://decomp.me/scratch/smg0X
The scracth context is old and needs to be updated to

Reply:
Not getting OK with that. Something else here I think